src/views/Task/Update.js

```diff
@@ -1,29 +1,192 @@
-import React from 'react'
+import React, { Component } from 'react'
 import Layout from '../../components/Layout'
+import task from '../../api/task'
+import DatePicker from '../../components/DatePicker'
+import moment from 'moment'
+import Alert from '../../components/Alert'
+import { withRouter } from 'react-router-dom'
+import camelcase from 'camelcase'
 
-export default () => (
-  <Layout>
-    <h2>Edit Task</h2>
-    <label>
-      Title
-      <input type="text" />
-    </label>
-    <label>
-      Start Date
-      <input type="date" />
-    </label>
-    <label>
-      Deadline
-      <input type="date" />
-    </label>
-    <label>
-      Description
-      <textarea />
-    </label>
-    <label>
-      Urgency
-      <input type="text" />
-    </label>
-    <input type="submit" value="Save Changes" />
-  </Layout>
-)
+export default withRouter(class Update extends Component {
+  state = {
+    startDate: moment(),
+    endDate: moment(),
+    title: "",
+    description: "",
+    urgency: "",
+    error: null
+  }
+
+  fields = [
+    //["label", "Type", Validation method]
+    ["Title", "text", v => v.length > 0, false],
+    ["Description", "text", v => v.length > 0, true],
+    ["Urgency", "text", v => v.length > 0, false]
+  ]
+
+  updateHandler = async event => {
+    event.preventDefault()
+    //invalid input handling
+    if (!this.validator()) {
+      this.setState({ error: "Invalid input" })
+      return
+    }
+
+    //input valid
+    try {
+      const taskData = {
+        title: this.state.title,
+        description: this.state.description,
+        urgency: this.state.urgency,
+        start: this.state.startDate.toDate(), //toDate() to convert moment()-date to standard JS-date, due to Superstruckt and server limitations
+        deadline: this.state.endDate.toDate() //toDate() to convert moment()-date to standard JS-date, due to Superstruckt and server limitations
+      }
+      await task.update(this.props.match.params.id, taskData)
+      this.props.history.push('/tasks')
+    } catch (e) {
+      this.setState({ error: e.message })
+    }
+  }
+
+  /**
+     * Validates input based on validation method defined in Fields[]
+     * @return {boolean}
+     */
+  validator = () => {
+    for (const [label, _, validator] of this.fields) { // eslint-disable-line
```

Comment:
'_' is never used in the method, can be deleted

Reply:
Nope can’t be deleted bevares er ser destructuring and er need the validator at index 2.